src/main/java/org/openrewrite/java/testing/sonar/TestsShouldIncludeAssertions.java

```diff
@@ -0,0 +1,155 @@
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.sonar;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.java.AnnotationMatcher;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.Statement;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * For Tests not having any assertions, wrap the statements with JUnit 5's Assertions.assertThrowDoesNotThrow.
+ *
+ * <a href="https://rules.sonarsource.com/java/tag/tests/RSPEC-2699">Sonar Source RSPEC-2699</a>
+ */
+@Incubating(since = "1.1.0")
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class TestsShouldIncludeAssertions extends Recipe {
+
+    private static final AnnotationMatcher JUNIT_JUPITER_TEST = new AnnotationMatcher("@org.junit.jupiter.api.Test");
+
+    private static final String THROWING_SUPPLIER_FQN = "org.junit.jupiter.api.function.ThrowingSupplier";
+    private static final String ASSERTIONS_FQN = "org.junit.jupiter.api.Assertions";
+    private static final String ASSERTIONS_DOES_NOT_THROW_FQN = "org.junit.jupiter.api.Assertions.assertDoesNotThrow";
+    private static final String ASSERT_DOES_NOT_THROW = "assertDoesNotThrow";
+
+    private static final ThreadLocal<JavaParser> ASSERTIONS_PARSER = ThreadLocal.withInitial(() ->
+            JavaParser.fromJavaVersion().dependsOn(Arrays.asList(
+                    Parser.Input.fromString(
+                            "package org.junit.jupiter.api.function;" +
+                                    "public interface ThrowingSupplier<T> {T get() throws Throwable;}"
+                    ),
+                    Parser.Input.fromString(
+                            "package org.junit.jupiter.api;" +
+                                    "import org.junit.jupiter.api.function.ThrowingSupplier;" +
+                                    "class AssertDoesNotThrow {" +
+                                    "   static <T> T assertDoesNotThrow(ThrowingSupplier<T> supplier) {\n" +
+                                    "       return (T)(Object)null;\n" +
+                                    "   }" +
+                                    "}"
+                    )
+            )).build());
+
+    @Option(displayName = "Assertion Packages",
+            description = "List of fully qualified package names of test assertion packages used for finding assertion statements.",
+            example = "org.junit.jupiter.api.Assertions, org.hamcrest.MatcherAssert")
+    List<String> assertionPackages;
+
+    @Option(displayName = "Assert Select Methods",
+            description = "List of fully qualified assertion method names used for finding assertion statements.",
+            example = "org.mockito.Mockito.verify, io.vertx.ext.unit.TestContext.verify")
+    List<String> assertMethods;
```

Comment:
These look really similar. What if it were a single configuration option that handled both fully qualified class and method names?

Also `assertionPackages` name is misleading since you populate it with fully qualified class names, not packages.